geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/JndiBindingConfiguration.java

```diff
@@ -0,0 +1,218 @@
+  private Integer maxPoolSize;
+  private String password;
+  private String transactionType;
+  private DATASOURCE_TYPE type;
+  private String username;
+  private String xaDatasource;
+
+  private List<ConfigProperty> dsConfigurations;
+
+  public JndiBindingConfiguration() {
+    dsConfigurations = new ArrayList<>();
+  }
+
+  public Integer getBlockingTimeout() {
+    return blockingTimeout;
+  }
+
+  public void setBlockingTimeout(Integer blockingTimeout) {
+    this.blockingTimeout = blockingTimeout;
+  }
+
+  public String getConnectionPoolDatasource() {
+    return connectionPoolDatasource;
+  }
+
+  public void setConnectionPoolDatasource(String connectionPoolDatasource) {
+    this.connectionPoolDatasource = connectionPoolDatasource;
+  }
+
+  public String getConnectionUrl() {
+    return connectionUrl;
+  }
+
+  public void setConnectionUrl(String connectionUrl) {
+    this.connectionUrl = connectionUrl;
+  }
+
+  public Integer getIdleTimeout() {
+    return idleTimeout;
+  }
+
+  public void setIdleTimeout(Integer idleTimeout) {
+    this.idleTimeout = idleTimeout;
+  }
+
+  public Integer getInitPoolSize() {
+    return initPoolSize;
+  }
+
+  public void setInitPoolSize(Integer initPoolSize) {
+    this.initPoolSize = initPoolSize;
+  }
+
+  public String getJdbcDriver() {
+    return jdbcDriver;
+  }
+
+  public void setJdbcDriver(String jdbcDriver) {
+    this.jdbcDriver = jdbcDriver;
+  }
+
+  public String getJndiName() {
+    return jndiName;
+  }
+
+  public void setJndiName(String jndiName) {
+    this.jndiName = jndiName;
+  }
+
+  public Integer getLoginTimeout() {
+    return loginTimeout;
+  }
+
+  public void setLoginTimeout(Integer loginTimeout) {
+    this.loginTimeout = loginTimeout;
+  }
+
+  public String getManagedConnFactory() {
+    return managedConnFactory;
+  }
+
+  public void setManagedConnFactory(String managedConnFactory) {
+    this.managedConnFactory = managedConnFactory;
+  }
+
+  public Integer getMaxPoolSize() {
+    return maxPoolSize;
+  }
+
+  public void setMaxPoolSize(Integer maxPoolSize) {
+    this.maxPoolSize = maxPoolSize;
+  }
+
+  public String getPassword() {
+    return password;
+  }
+
+  public void setPassword(String password) {
+    this.password = password;
+  }
+
+  public String getTransactionType() {
+    return transactionType;
+  }
+
+  public void setTransactionType(String transactionType) {
+    this.transactionType = transactionType;
+  }
+
+  public DATASOURCE_TYPE getType() {
+    return type;
+  }
+
+  public void setType(DATASOURCE_TYPE type) {
+    this.type = type;
+  }
+
+  public String getUsername() {
+    return username;
+  }
+
+  public void setUsername(String username) {
+    this.username = username;
+  }
+
+  public String getXaDatasource() {
+    return xaDatasource;
+  }
+
+  public void setXaDatasource(String xaDatasource) {
+    this.xaDatasource = xaDatasource;
+  }
+
+  public List<ConfigProperty> getDsConfigurations() {
```

Comment:
Please make this `getDatasourceConfigurations` - 'DS' seems to already be used as an abbreviation in various places in our code - DistributedSystem, DataSerializer, etc.

Reply:
Done.